MODRTAC-89: Fetch of holdings records fails for mixed instances

K-Felk opened the pull request:
I traced this problem to the part of the code that looks up circulation
data for any items attached to the requested instance records.  Because
of the way this code had been written, it effectively discarded any 
holdings information for instance records that did not have attached items.
Later code flags any records that don't have holdings data as 404 not found,
making it appear that the holdings data could not be retrieved.

To fix, I made a list of instance records without attached items, then merge this
list in later with the list of items that do, and return the merged list to the 
calling code.

marcjohnson-kint commented on src/main/java/org/folio/clients/CirculationClient.java:
```diff
@@ -65,7 +70,12 @@
     CompositeFuture.all(futures)
-        .onSuccess(updatedInstances -> promise.complete(updatedInstances.result().list()))
+        .onSuccess(updatedInstances -> {
+          List<InventoryHoldingsAndItems> combinedList = new ArrayList<>();
+          combinedList.addAll(updatedInstances.result().list());
+          combinedList.addAll(instancesNoItems);
+          promise.complete(combinedList);
```
Could an alternative for this be to not remove the instances without items from the list in the first place?

marcjohnson-kint reviewed (commented):
This change has no tests to replicate the issue. 

Are we intending to expedite it so it can be tested on an environment where this can be tested? (and then come back and add tests for this later)

marcjohnson-kint commented:
@K-Felk  Why was this merged when the sonar checks / branch builds were still failing?

K-Felk commented:
Two reasons:
1.  Code failures remaining were in parts of the code I had not touched.  I removed all code smells from the code I had changed and updated.
2. This part of the check is not marked as required, leading me to believe it could be merged.

I don't think I've ever had this part of the process fail, so if this was not the way to handle it, then I apologize.

marcjohnson-kint commented:
> This part of the check is not marked as required, leading me to believe it could be merged.

That is a good question.

@funkymalc @wafschneider  Any ideas why the builds of this pull request would be marked as failures and yet the pull request could be still merged?

Is it because the sonar checks are optional and thus don't have to pass for the build to be considered ok (despite getting a red cross)?
